site/app/libraries/Access.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+
+namespace app\libraries;
+
+
+use app\models\Gradeable;
+use app\models\User;
+
+class Access {
+    const USER_GROUP_INSTRUCTOR = 1;
+    const USER_GROUP_TA         = 2;
+    const USER_GROUP_GRADER     = 3;
+    const USER_GROUP_STUDENT    = 4;
+
+    // Bit field for various access control options
+    const ALLOW_INSTRUCTOR              = 1 << 0;
+    const ALLOW_TA                      = 1 << 1;
+    const ALLOW_GRADER                  = 1 << 2;
+    const ALLOW_STUDENT                 = 1 << 3;
+    const ALLOW_LOGGED_OUT              = 1 << 4;
+    const CHECK_GRADEABLE_MIN_GROUP     = 1 << 5;
+    const CHECK_GRADING_SECTION_GRADER  = 1 << 6;
+    const CHECK_PEER_ASSIGNMENT_STUDENT = 1 << 7;
+    const CHECK_HAS_SUBMISSION          = 1 << 8;
+    const CHECK_CSRF                    = 1 << 9;
+
+    //
+    const ALLOW_MIN_STUDENT    = self::ALLOW_INSTRUCTOR | self::ALLOW_TA | self::ALLOW_GRADER | self::ALLOW_STUDENT;
```

Comment:
What's the point of this check?

Reply:
Mostly consistency with the other access levels so it's obvious what access level is required. Also future-proofing in case we ever decide to add more user groups or do something different.